core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -691,20 +727,29 @@ public FileInfo getFileInfo(long fileId)
   public FileInfo getFileInfo(AlluxioURI path, GetStatusOptions options)
       throws FileDoesNotExistException, InvalidPathException, AccessControlException {
     Metrics.GET_FILE_INFO_OPS.inc();
-
     try (JournalContext journalContext = createJournalContext();
-         LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.READ)) {
-      mPermissionChecker.checkPermission(Mode.Bits.READ, inodePath);
+         LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.READ);
+         MasterAuditContext auditContext = createAuditContext("getFileInfo", path, null, null)) {
+      try {
+        mPermissionChecker.checkPermission(Mode.Bits.READ, inodePath);
+      } catch (AccessControlException e) {
+        auditContext.setAllowed(false);
+        throw e;
+      }
       if (inodePath.fullPathExists()) {
```

Comment:
can we restructure the code a little so that there's only one `return` point?
```java
if (!inodePath.fullPathExists()) {
  loadMetadataEtc
  ...
}
FileInfo fileInfo = getFileInfoInternal(inodePath);
auditContext.setSrcInode(inodePath.getInode()).setSuccess(true);
return fileInfo;
```